core/field_textinput.js

```diff
@@ -295,34 +295,33 @@ Blockly.FieldTextInput.prototype.widgetCreate_ = function() {
 };
 
 /**
- * Close the editor, save the results, and dispose any events bound to the
- * text input's editor.
+ * Closes the editor, saves the results, and disposes of any events or
+ * dom-references belonging to the editor.
  * @private
  */
 Blockly.FieldTextInput.prototype.widgetDispose_ = function() {
-  // Finalize value.
+  this.onEditorClose_();
+  this.unbindInputEvents_();
+  var style = Blockly.WidgetDiv.DIV.style;
+  style.width = 'auto';
```

Comment:
The widget div should probably do that, similar to how the dropdown div clears itself between uses.

Reply:
Ok I moved the responsibility for css into dropdown div and widget div.

It may break people who are depending on how we currently handle the editor css though, so I'd love opinions.

Also I need to verify that none of the dropdown div issues from #2551 have come back.